accepted/0000-npm7-support-package-lock-6.md

```diff
@@ -0,0 +1,59 @@
+# Support `package-lock.json` v3 in npm 7
+
+## Summary
+
+Currently npm 7 can read `package-lock.json` versions 1, 2, and 3, but it always writes version 3. It would be nice if npm 7 could write `package-lock.json` version 3 instead.
+
+## Motivation
+
+npm 7 introduces new features which aren’t compatible with older npm versions, such as workspaces. This makes npm 7 a minimum requirement for a project.
+
+Both `package-lock.json` version 1 and 3 files can be huge. `package-lock.json` 2 is an intermediate format whose goal is to add backwards compatibility with `package-lock.json` version 1 and npm 6. It does so by writing both the dependency trees in the version 1 and 3 formats. This means the file is even twice as big as it needs to be. This also means the same change appears twice in a diff, but in a different format, meaning it needs to be reviewed twice. This is unnecessary if the minimal required npm version is 7.
```

Comment:
```suggestion
Both `package-lock.json` version 1 and 2 files can be huge. `package-lock.json` 2 is an intermediate format whose goal is to add backwards compatibility with `package-lock.json` version 1 and npm 6. It does so by writing both the dependency trees in the version 1 and 3 formats. This means the file is even twice as big as it needs to be. This also means the same change appears twice in a diff, but in a different format, meaning it needs to be reviewed twice. This is unnecessary if the minimal required npm version is 7.
```
?

Reply:
Nope, I meant versions 1 and 3. Version 2 includes both, which makes it even bigger.